azurelinuxagent/ga/update.py

```diff
@@ -980,6 +985,34 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    @staticmethod
+    def _add_nonroot_tcp_firewallrule_if_not_enabled(dst_ip):
+
+        update_handler = get_update_handler()
+
+        # "-C" checks if the iptable rule is available in the chain
+        drop_rule = _get_firewall_drop_command("-w", "-C", dst_ip)
+        if not update_handler._check_if_iptable_rule_is_available(drop_rule):
+            return
+        else:
+            accept_non_root = _get_firewall_accept_command_nonroot_tcp("-w", "-C", dst_ip)
+            if not update_handler._check_if_iptable_rule_is_available(accept_non_root):
+                try:
+                    logger.info("Firewall rule to allow non root users to do tcp request to wireserver unavailable . Setting it now.")
+                    accept_non_root = _get_firewall_accept_command_nonroot_tcp("-w", "-I", dst_ip)
+                    shellutil.run_command(accept_non_root)
+                    logger.info("Succesfully added firewall rule to allow non root users to do tcp request to wireserver ")
+                except Exception as e:
+                    msg = "Unable to set the nonroot tcp access firewall rule:{0}".format(ustr(e))
+                    logger.error(msg)
+
+    @staticmethod
+    def _check_if_iptable_rule_is_available(command):
```

Comment:
we need to rename this function: it has nothing to do with iptables, it just executes an arbitrary command

Reply:
Additionally, just noticed this is only used by the `_add_nonroot_tcp_firewallrule_if_not_enabled` function. Maybe consider making it an inline function inside that function since its localized?